components/incidents/createIncidents/CreateIncident.tsx

```diff
@@ -0,0 +1,363 @@
+        <IncidentName value = {incidentName} handleNameChange = {(e:React.BaseSyntheticEvent) => handleNameChange(e)}/>
+        <InputStatus updateStatus = {(e:React.BaseSyntheticEvent) => updateStatus(e)} incidentStatus = {incidentStatus} updateStatusBarOnClick = {(event:SpecialEvent) => updateStatusBarOnClick(event)}/>
+        <IncidentMessage value = {incidentMessage} updateIncidentMessage = {(e:React.BaseSyntheticEvent) => updateIncidentMessage(e)}/>  
+    </>);
+    if(isLoaded != 2){
+        if(!isSubmitClicked)
+            return (
+                <>
+                <div className = {styles.main}>
+                    {formConstant}
+                    {isLoaded == 1 ? <><ComponentsAffected componentList = {componentsAffected} toggleCheckBox = {(e:React.BaseSyntheticEvent) => toggleCheckBox(e)} changeOption ={(e:optionType, id:String) => changeOption(e, id)}/> 
+                    <Button
+                    onClick={() => {
+                    setIsSubmitClicked(true);
+                    submitForm();
+                    // dequeue();
+                    enqueue({
+                        message: 'Submitting Form Details',
+                        progress: true
+                    }, DURATION.infinite)}} 
+                    overrides={{BaseButton : {style: ({$theme}) => ({backgroundColor: $theme.colors.accent,width: '80px',alignSelf: 'end'})}}}>
+                        Create
+                    </Button></> 
+                    : <div className={styles.Spinner}><Spinner $size={SIZE.large} /></div> }
+                    </div></>
+            );
+        else{
+            return (
+                <>
+                <div className = {styles.main}>
+                    {formConstant}
+                    {isLoaded == 1 ? <><ComponentsAffected componentList = {componentsAffected} toggleCheckBox = {(e:React.BaseSyntheticEvent) => toggleCheckBox(e)} changeOption ={(e:optionType, id:String) => changeOption(e, id)}/> 
+                    <Button
+                    overrides={{BaseButton : {style: ({$theme}) => ({backgroundColor: $theme.colors.accent,width: '80px',alignSelf: 'end', cursor: "wait"})}}}>
+                        Create
+                    </Button></> 
+                    : <div className={styles.Spinner}><Spinner $size={SIZE.large} /></div> }
+                    </div></>
+            );
+        }
+    }
+    else{
+        return (
+            <>
+            <div className = {styles.main}>
+                {formConstant}
+                <Block overrides={
+                    {
+                        Block:{
+                            style: {
+                                display: "flex",
+                                alignContent: "center",
+                                justifyContent: "center"
+                            }
+                        }
+                    }
+                }>
+                    <h1>Sorry Unable to Fetch Components</h1>
```

Comment:
should we even show form in this case ?

Reply:
Done. Now only only showing not able to fetch components on failed fetch